kolibri/core/assets/src/state/store.js

```diff
@@ -170,4 +174,17 @@ const mutations = {
   },
 };
 
-export { initialState, mutations };
+const store = new Vuex.Store({});
+
+export default store;
+
+store.registerModule = ({ state, mutations } = { state: {}, mutations: {} }) => {
+  if (store.__initialized) {
+    throw new Error(
+      'The store has already been initialized, dynamic initalization is not currently available'
+    );
+  }
+  store.hotUpdate({ mutations: Object.assign(mutations, coreMutations) });
```

Comment:
Tacking `registerModule` on to the existing store feels a bit hacky to me, in particular the use of the delayed `hotUpdate`. It's a good nod to vuex2, but I'm not sure that it gains us that much?

It might be cleaner to defer instantiation of the `new Vuex.Store()` until the [`ready` method of `kolibri_app`](https://github.com/learningequality/kolibri/pull/2540/files#diff-ac142500181ee23f6bc0272fdeadcac8R31).

Reply:
Yeah, I did actually try that first, but it made it very difficult to then have a consistent reference to it from within the core app itself. Being able to require it (as this allows) made that work, whereas other attempts to reference it from within the core app either failed due to reference errors, or failure to update the module referent when I tried to instantiate it dynamically.